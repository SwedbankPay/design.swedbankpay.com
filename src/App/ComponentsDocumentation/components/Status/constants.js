import React from "react";

const Status = ({ right }) => (
    <ul className={`list status-list d-flex flex-column${right ? " align-items-end" : ""}`}>
        <li>
            <span className={`status${right ? " status-right" : ""} status-neutral`}>Neutral</span>
        </li>
        <li>
            <span className={`status${right ? " status-right" : ""} status-success`}>Success</span>
        </li>
        <li>
            <span className={`status${right ? " status-right" : ""} status-warning`}>Warning</span>
        </li>
        <li>
            <span className={`status${right ? " status-right" : ""} status-danger`}>Danger</span>
        </li>
        <li>
            <span className={`status${right ? " status-right" : ""} status-inactive`}>Inactive</span>
        </li>
    </ul>
);

export const showcaseStatusComponent = {
    id: "overviewStatus",
    elements: [
        {
            component: <Status/>,
            options: {
                checkbox: [
                    {
                        title: "Position",
                        inputs: [
                            {
                                id: "right",
                                name: "Trailing",
                                value: {
                                    right: true
                                }
                            }
                        ]
                    }
                ]
            },
            title: "Status",
            description: "The component has two different states; leading and trailing. Use the status most suitable for your solution. Please beware, when using the trailing option, you must self-add code to make it align correctly."
        }
    ]
};

export const leadingExample = (
    <table className="table table-plain">
        <thead>
            <tr>
                <th scope="col">Status</th>
                <th scope="col">Invoice number</th>
                <th scope="col">Due date</th>
                <th className="text-right" scope="col">Amount</th>
            </tr>
        </thead>
        <tbody>
            <tr>
                <td><span className="status status-neutral">Unpaid</span></td>
                <td>123456</td>
                <td>2020-08-19</td>
                <td className="text-right">1079,00</td>
            </tr>
            <tr>
                <td><span className="status status-success">Paid</span></td>
                <td>123455</td>
                <td>2020-07-19</td>
                <td className="text-right">1945,00</td>
            </tr>
            <tr>
                <td><span className="status status-warning">Overdue</span></td>
                <td>123454</td>
                <td>2020-06-19</td>
                <td className="text-right">1814,00</td>
            </tr>
            <tr>
                <td><span className="status status-danger">Sent to collection</span></td>
                <td>123452</td>
                <td>2020-04-19</td>
                <td className="text-right">1337,00</td>
            </tr>
            <tr>
                <td><span className="status status-inactive">Reserved</span></td>
                <td>123453</td>
                <td>1994-02-22</td>
                <td className="text-right">69 420,00</td>
            </tr>
        </tbody>
    </table>
);

export const trailingExample = (
    <table className="table table-plain">
        <thead>
            <tr>
                <th scope="col">Name</th>
                <th className="text-right" scope="col">Status</th>
            </tr>
        </thead>
        <tbody>
            <tr>
                <td>Micheal Scott</td>
                <td className="d-flex align-items-center justify-content-end"><span className="status status-right status-neutral">Current Manager</span></td>
            </tr>
            <tr>
                <td>Dwight Shrute</td>
                <td className="d-flex align-items-center justify-content-end"><span className="status status-right status-success">Assistant (to the) manager</span></td>
            </tr>
            <tr>
                <td>Jim Halpert</td>
                <td className="d-flex align-items-center justify-content-end"><span className="status status-right status-warning">Time waster</span></td>
            </tr>
            <tr>
                <td>Todd Packer</td>
                <td className="d-flex align-items-center justify-content-end"><span className="status status-right status-danger">Former employee</span></td>
            </tr>
            <tr>
                <td>Pam Beesly</td>
                <td className="d-flex align-items-center justify-content-end"><span className="status status-right status-inactive">No experience</span></td>
            </tr>
        </tbody>
    </table>
);
