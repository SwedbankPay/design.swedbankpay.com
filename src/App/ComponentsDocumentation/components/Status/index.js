import React from "react";

import { ComponentPreview, DocContainer } from "@docutils";
import Alert from "@components/Alert";
import CodeTags from "@components/CodeTags";

const Overview = () => (
    <>
        <h2 id="overview">Overview</h2>
        <p>
            Be aware of what color you use when deciding what message you want to convey. A completion status should not use a warning or danger status,
            that is why <CodeTags type="secondary" code=".status-success" /> exists.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure removeList>
            <ul className="list">
                <li>
                    <span className="status status-default">Default</span>
                </li>
                <li>
                    <span className="status status-neutral">Neutral</span>
                </li>
                <li>
                    <span className="status status-success">Success</span>
                </li>
                <li>
                    <span className="status status-warning">Warning</span>
                </li>
                <li>
                    <span className="status status-danger">Danger</span>
                </li>
            </ul>
        </ComponentPreview>
        <Alert type="warning">
            <h5>Visual impairments and colors</h5>
            <p>
                Be careful not to rely soley on the color of the status icon to convey your message.
            </p>
        </Alert>
    </>
);

const ExtendedUsage = () => (
    <>
        <h2 id="extended-usage">Extended usage</h2>
        <p>
            The status component inherits the size of its direct parent.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <h1 className="status status-default">Default</h1>
            <h2 className="status status-neutral">Neutral</h2>
            <h3 className="status status-success">Success</h3>
            <h4 className="status status-warning">Warning</h4>
            <h5 className="status status-danger">Danger</h5>
        </ComponentPreview>
    </>
);

const Status = () => (
    <DocContainer docToc>
        <p className="lead">
            The status component is a good way to convey to the user the state of something.
        </p>
        <Overview />
        <ExtendedUsage />
    </DocContainer>
);

export default Status;

/* For testing */
export { Overview, ExtendedUsage };
