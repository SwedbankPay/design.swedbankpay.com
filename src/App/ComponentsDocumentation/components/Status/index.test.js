import React from "react";
import { shallow } from "enzyme";

import Status, { Overview, ExampleUse, ExtendedUsage } from "./index";

describe("Documentation: Status", () => {
    it("is defined", () => {
        expect(Status).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<Status />);

        expect(wrapper).toMatchSnapshot();
    });

    describe("Overview", () => {
        it("is defined", () => {
            expect(Overview).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<Overview />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("ExampleUse", () => {
        it("is defined", () => {
            expect(ExampleUse).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<ExampleUse />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("ExtendedUsage", () => {
        it("is defined", () => {
            expect(ExtendedUsage).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<ExtendedUsage />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
