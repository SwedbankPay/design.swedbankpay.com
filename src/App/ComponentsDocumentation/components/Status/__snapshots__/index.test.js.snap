// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documentation: Status ExtendedUsage renders 1`] = `
<Fragment>
  <h2
    id="extended-usage"
  >
    Extended usage
  </h2>
  <p>
    The status component inherits the size of its direct parent.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <h1
      className="status status-default"
    >
      Default
    </h1>
    <h2
      className="status status-neutral"
    >
      Neutral
    </h2>
    <h3
      className="status status-success"
    >
      Success
    </h3>
    <h4
      className="status status-warning"
    >
      Warning
    </h4>
    <h5
      className="status status-danger"
    >
      Danger
    </h5>
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: Status Overview renders 1`] = `
<Fragment>
  <h2
    id="overview"
  >
    Overview
  </h2>
  <p>
    Be aware of what color you use when deciding what message you want to convey. A completion status should not use a warning or danger status, that is why 
    <CodeTags
      code=".status-success"
      type="secondary"
    />
     exists.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    removeList={true}
    showCasePanel={true}
  >
    <ul
      className="list"
    >
      <li>
        <span
          className="status status-default"
        >
          Default
        </span>
      </li>
      <li>
        <span
          className="status status-neutral"
        >
          Neutral
        </span>
      </li>
      <li>
        <span
          className="status status-success"
        >
          Success
        </span>
      </li>
      <li>
        <span
          className="status status-warning"
        >
          Warning
        </span>
      </li>
      <li>
        <span
          className="status status-danger"
        >
          Danger
        </span>
      </li>
    </ul>
  </ComponentPreview>
  <Alert
    type="warning"
  >
    <h5>
      Visual impairments and colors
    </h5>
    <p>
      Be careful not to rely soley on the color of the status icon to convey your message.
    </p>
  </Alert>
</Fragment>
`;

exports[`Documentation: Status renders 1`] = `
<DocContainer
  docToc={true}
>
  <p
    className="lead"
  >
    The status component is a good way to convey to the user the state of something.
  </p>
  <Overview />
  <ExtendedUsage />
</DocContainer>
`;
